File: front/ProyectoTesis/src/app/models/mejor-ruta-response.dto.ts
export interface ParaderoDTO {
  id: number;
  nombre: string;
  lat: number;
  lng: number;
}

export interface MejorRutaResponseDTO {
  mensaje?: string;
  origen?: ParaderoDTO;
  destino?: ParaderoDTO;
  rutaId?: number;
  rutaNombre?: string;
  sentido?: string;
  distanciaOrden?: number;
  paraderosRuta?: ParaderoDTO[];
}

File: back/ProyectoTesis/src/main/java/com/ecapi/dto/MejorRutaResponseDTO.java
package com.ecapi.dto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class MejorRutaResponseDTO {

	private String mensaje;
	private ParaderoDTO origen;
	private ParaderoDTO destino;
	private Long rutaId;
	private String rutaNombre;
	private String sentido;
	private Integer distanciaOrden;

	// Para trazar la ruta en el mapa private
	List<ParaderoDTO> paraderosRuta;
}

